Feat: New DNS client for Gateway 3.8 (#7805)
Kong/docs.konghq.com: app/_src/gateway/migrate-to-new-dns-client.md

---

> title: Migrate to the new DNS client

Reviewer: I think maybe this would fit better in the Networking section of the docs (same section as https://deploy-preview-7805--kongdocs.netlify.app/gateway/latest/production/networking/dns-considerations/#main).

I can't decide if I think this would be better if it was added to that existing DNS doc. I searched for CNAME and SRV in that other DNS doc and didn't come up with anything. Is there something we could add from this doc to that one?

Reviewer: > maybe we should have both? I could add a mention of the two clients in the DNS considerations section.

I think this sounds good to me!

---

> This guide walks you through migrating from the old DNS client to the new one. The new DNS client introduces and new standardized way to configure a service, and helps improve performance.
> 

Reviewer 2: maybe ?
```suggestion
This guide walks you through migrating from the old DNS client to the new one. The new DNS client introduces a new standardized way to configure a service, and helps improve performance.

```

Author: ah yes, thanks!

> ## Migrate the DNS client configuration

Reviewer: I'm not super clear on the migration steps. Do I change the custom directives over and then enable the new DNS client?

Author: I guess you would probably do everything at the same time, I added this section first because you shouldn't enable the client before at least checking the rest of the config

Reviewer: This makes sense to me then!

### Record types

> The new DNS client introduces some changes in the support for different record types. To avoid issues, make sure to check your configuration before migrating.

Reviewer 3: ```suggestion
The new DNS client introduces some changes in the support for different record types. To avoid issues, make sure to check your configuration before migrating.
```

Is there anything specific that I should be checking for? Record types?

Author: Yes, mainly the domain name for SRV records

Reviewer 3: Write that specifically here.

#### SRV

> The new DNS client handles SRV records differently than the previous one. If you enable SRV support with the `resolver_family` directive, the client will only query SRV records if the domain name follows the RFC 2782 format (`_<service>._<proto>.<name>`). If the SRV record query fails, the client will not attempt to query the domain's IP addresses (A and AAAA records) again.

Reviewer 2: yes SRV is in `resolver_family` by default

> The new DNS client handles SRV records differently than the previous one. If you enable SRV support with the `resolver_family` directive, the client will only query SRV records if the domain name follows the RFC 2782 format (`_<service>._<proto>.<name>`). If the SRV record query fails, the client will not attempt to query the domain's IP addresses (A and AAAA records) again.
> 
> Before enabling SRV support with the new DNS client, make sure that the domain name is registered with your DNS service provider in the supported format.

Reviewer 3: ```suggestion
If you enable SRV support with the `resolver_family` directive, the client will only query SRV records if the domain name follows the RFC 2782 format (`_<service>._<proto>.<name>`). If the SRV record query fails, the client will not attempt to query the domain's IP addresses (A and AAAA records) again.

Before enabling SRV support with the new DNS client, make sure that the domain name is registered with your DNS service provider in the supported format.
```

#### CNAME

> The new DNS client no longer performs CNAME dereferencing; this task is entirely handled by the DNS server.
> 
> {:.note}
> > The new DNS client does not consider the order of record types when querying a domain. It only queries either IP addresses (A and AAAA records) or SRV records, but not both.

Reviewer 3: ```suggestion
The DNS client does not offer CNAME dereferencing, this task is entirely handled by the DNS server.

{:.note}
> The new DNS client does not consider the order of record types when querying a domain. It only queries either IP addresses (A and AAAA records) or SRV records, but not both.
```
